### PR TITLE
[react-dom] Add client exports

### PR DESCRIPTION
**Pull request by Brianzchen**
Add type defs for `react-dom/client`

- Links to documentation: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html
- Link to GitHub or NPM: https://www.npmjs.com/package/react-dom
- Type of contribution: addition

Other notes:

### CODE REVIEW

**Comment by pascalduez** [comment]
**question:** Any reason to not have copied the sources types? 
Which BTW is a bit of a pity that they are not published with the package...  

https://github.com/facebook/react/blob/168da8d55782f3b34e2a6aa0c4dd0587696afdbd/packages/react-dom/src/client/ReactDOM.js#L147-L176

https://github.com/facebook/react/blob/168da8d55782f3b34e2a6aa0c4dd0587696afdbd/packages/react-dom/src/client/ReactDOMRoot.js#L19-L24

https://github.com/facebook/react/blob/168da8d55782f3b34e2a6aa0c4dd0587696afdbd/packages/shared/ReactTypes.js#L22

**Comment by Brianzchen** [comment]
> **question:** Any reason to not have copied the sources types?

No reason at all! I will do that instead, updates coming when I get a chance

File: definitions/npm/react-dom_v18.x.x/flow_v0.127.x-/react-dom_v18.x.x.js [hunk 1]
```diff
@@ -43,6 +146,56 @@ declare module 'react-dom' {
...
+    identifierPrefix?: string,
+    onRecoverableError?: (error: mixed) => void,
+    ...
+  };
+
+  declare export function hydrateRoot<ElementType: React$ElementType>(
+    container: Document | Element,
+    initialChildren: ReactNodeList,
+    options?: HydrateRootOptions,
+  ): void;
```
Comment (pascalduez) [inline definitions/npm/react-dom_v18.x.x/flow_v0.127.x-/react-dom_v18.x.x.js:196]:
```suggestion
  ): RootType;
```

File: definitions/npm/react-dom_v18.x.x/flow_v0.127.x-/react-dom_v18.x.x.js [hunk 2]
```diff
@@ -43,6 +146,56 @@ declare module 'react-dom' {
...
+    onDeleted?: (suspenseNode: Comment) => void,
+    // Options for all roots
+    unstable_strictMode?: boolean,
+    unstable_concurrentUpdatesByDefault?: boolean,
+    identifierPrefix?: string,
+    onRecoverableError?: (error: mixed) => void,
+    ...
+  };
+
+  declare export function hydrateRoot<ElementType: React$ElementType>(
```
Comment (pascalduez) [inline definitions/npm/react-dom_v18.x.x/flow_v0.127.x-/react-dom_v18.x.x.js:192]:
```suggestion
  declare export function hydrateRoot(
```

**Review by pascalduez — APPROVED** [review]

**Pull request merged**